src/webapp/components/import-translation-dialog/ImportTranslationDialog.tsx

```diff
@@ -1,109 +1,73 @@
 import { ConfirmationDialog, Dropdown, DropdownItem, useSnackbar } from "@eyeseetea/d2-ui-components";
-import _ from "lodash";
-import React, { useCallback, useImperativeHandle, useRef, useState } from "react";
+import React, { useCallback, useState } from "react";
 import styled from "styled-components";
 import { Action } from "../../../domain/entities/Action";
-import { TranslateMethod } from "../../../domain/entities/TranslatableText";
+import { TranslateMethod, Translations } from "../../../domain/entities/TranslatableText";
 import i18n from "../../../utils/i18n";
 import { useAppContext } from "../../contexts/app-context";
 
-export const ImportTranslationDialog = React.forwardRef(
-    (props: ImportTranslationDialogProps, ref: React.ForwardedRef<ImportTranslationRef>) => {
-        const { actions, translate } = useAppContext();
-        const snackbar = useSnackbar();
-
-        const [open, setOpen] = useState<boolean>(false);
-        const [selectedLang, setSelectedLang] = useState<string>();
-        const [selectedAction, setSelectedAction] = useState<string>();
-        const [terms, setTerms] = useState<Record<string, string>>();
-
-        const inputRef = useRef<any>(null);
-
-        const save = useCallback(async () => {
-            if (props.type === "action" && !selectedAction) {
-                snackbar.error(i18n.t("You need to select an action"));
-                return;
-            }
-
-            if (!selectedLang || !terms) {
-                snackbar.error(i18n.t("You need to select a language"));
-                return;
-            }
-            await props.onSave(selectedAction, selectedLang, terms);
-
-            setOpen(false);
-        }, [snackbar, props, selectedAction, selectedLang, terms]);
-
-        const onFileUpload = useCallback(
-            async (event: any) => {
-                try {
-                    const file = event.target.files[0];
-                    if (!file) throw new Error("No file received on upload");
-
-                    const text = await file.text();
-                    const json = JSON.parse(text);
-                    const terms = _.pickBy(json, value => _.isString(value));
+export type ImportTranslationDialogProps = {
+    type: "action" | "landing-page" | "notification";
+    onSave: (key: string | undefined, lang: string, terms: Record<string, string>) => void | Promise<void>;
```

Comment:
[No need to change] For the record, `key: string | undefined, lang: string, terms: Record<string, string>` is a group of data that's used more than once. That typically means that there is an object type to be created so we "unify" them as a single concept/argument/response.